pkg/deploy/util/util.go

```diff
@@ -210,6 +226,53 @@ func HasImageChangeTrigger(config *deployapi.DeploymentConfig) bool {
 	return false
 }
 
+// HasTrigger returns whether the provided deployment configuration has any trigger
+// defined or not.
+func HasTrigger(config *deployapi.DeploymentConfig) bool {
+	return HasChangeTrigger(config) || HasImageChangeTrigger(config)
+}
+
+// HasLastTriggeredImage returns whether all image change triggers in provided deployment
+// configuration has the lastTriggerImage field set (iow. all images were updated for
+// them). Returns false if deployment configuration has no image change trigger defined.
+func HasLastTriggeredImage(config *deployapi.DeploymentConfig) bool {
+	hasImageTrigger := false
+	for _, trigger := range config.Spec.Triggers {
+		if trigger.Type == deployapi.DeploymentTriggerOnImageChange {
+			hasImageTrigger = true
+			if len(trigger.ImageChangeParams.LastTriggeredImage) == 0 {
+				return false
+			}
+		}
+	}
+	return hasImageTrigger
+}
+
+// IsInitialDeployment returns whether the deployment configuration is the first version
+// of this configuration.
+func IsInitialDeployment(config *deployapi.DeploymentConfig) bool {
+	return config.Status.LatestVersion == 0
+}
+
+// RecordConfigChangeCause sets a deployment config cause for config change.
+func RecordConfigChangeCause(config *deployapi.DeploymentConfig) {
+	config.Status.Details = new(deployapi.DeploymentDetails)
+	config.Status.Details.Causes = []deployapi.DeploymentCause{{
+		Type: deployapi.DeploymentTriggerOnConfigChange,
+	}}
+	config.Status.Details.Message = "config change"
+}
+
+// RecordImageChangeCause sets a deployment config cause for image change.
+func RecordImageChangeCause(config *deployapi.DeploymentConfig, name string) {
+	config.Status.Details = new(deployapi.DeploymentDetails)
+	config.Status.Details.Causes = []deployapi.DeploymentCause{{
+		Type:         deployapi.DeploymentTriggerOnImageChange,
+		ImageTrigger: &deployapi.DeploymentCauseImageTrigger{From: api.ObjectReference{Kind: "DockerImage", Name: name}},
```

Comment:
@kargakis i guess this is ok, but just to be sure, this will always record the docker image (not image stream)... to do that I will have to search the triggers and find the trigger with lastTriggeredImage matching the name... can do that but don't want to complicate this if we can avoid it. wdyt?